WalletWasabi/Blockchain/Analysis/FeesEstimation/FeeProviders.cs

```diff
@@ -6,37 +6,33 @@ namespace WalletWasabi.Blockchain.Analysis.FeesEstimation
 {
 	public class FeeProviders : IFeeProvider, IDisposable
 	{
-		private AllFeeEstimate _allFeeEstimate;
-
-		public FeeProviders(IEnumerable<IFeeProvider> feeProviders)
+		public FeeProviders(IFeeProvider baseProvider, params IFeeProvider[]? feeProviders)
 		{
-			Providers = feeProviders;
 			Lock = new object();
 
-			SetAllFeeEstimate();
+			Providers = new List<IFeeProvider>();
+
+			if (feeProviders is { })
+			{
+				Providers.AddRange(feeProviders.Where(x => x is { }));
+			}
+
+			// Backend(synchronizer) fee provider is always the last provider.
+			Providers.Add(baseProvider);
```

Comment:
I don't understand why you add this here just to skip it in the for loop later. It does not make sense. Why not simply add a new property for that provider?

Reply:
I want to handle it the same as other providers, for example when subscribing/unsubscribing to thems events.

Reply:
IMO this is just a matter of logic, here - both can work, so I stick with the original concept.

Reply:
I see. Ok.